### PR TITLE
Updates for W3S at scale and adding architecture (that page needs work and diagram). 

### PR DESCRIPTION
**Pull request by non-fungible-nelson**
[Preview build](https://doc-web3signer-git-fork-non-fungible-nelson-main-infura-web.vercel.app/how-to/run-at-scale)

### CODE REVIEW

File: docs/concepts/architecture.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+The remote signer [loads private keys](../how-to/load-keys.md) into memory and responds to signature requests. If you are using an [HSM](../how-to/store-keys/hsm/_category_.json) or a [vault](../how-to/store-keys/vaults/_category_.json) for Execution Layer signing, the keys remain at rest. This component communicates with the slashing database, the APIs, and with the keystore, if used, to coordinate remote signing. 
+
+## The Slashing Database
+
+The [slashing database](./slashing-protection.md) Postgres database keeps track of what messages have been signed and by what key. Database locking ensures that if Web3Signer instances load the same keys, only one Web3Signer instance actually signs. 
+
+## The APIs
+
+Web3Signer supports REST and [JSON-RPC APIs](../reference/api/_category_.json) to sign consensus layer and execution layer payloads respectively. These connections should be carefully secured. Web3Signer offers [TLS communication](../how-to/configure-tls.md).
```
Comment (joaniefromtheblock) [inline docs/concepts/architecture.md:24]: "These connections should be carefully secured." Should we use the term "connections" for this? Could "requests", "ports", or something else work here?

File: docs/how-to/run-at-scale.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+## Threading Model Optimization
+
+Web3Signer's threading framework, [Vertx](https://vertx.io/docs/vertx-core/java/), while powerful, is not suitable for all environments without configuration. If you are experiencing request latency or blocked threads, consider adjusting the [worker pool size](../reference/cli/options.md#vertx-worker-pool-size).
+
+* **Concurrency Management**: Tailor the size of Web3Signer's thread pool to the expected load, preventing bottlenecks. If you see decreased attestation performance while the signing load is at its peak, increase the pool size.
+* **Dynamic Adjustments**: Implement monitoring tools to adjust threads in real-time based on current demand and workload. You should measure spikes and adjust the pool accordingly. We provide two [metrics](./monitor/metrics.md) that may help here: 
+    * http_vertx_worker_queue_delay: time spent by requests in queues before being processed
+    * http_vertx_worker_pool_completed_total : number of queries processed by Web3Signer
+
+## Load Balancing
```
Comment (joaniefromtheblock) [inline docs/how-to/run-at-scale.md:34]:
```suggestion
full
```

or 

```suggestion
hello
```

**Comment by joaniefromtheblock** [comment]
Hi @non-fungible-nelson. Made a couple of changes. Could you please review?

**Review by joaniefromtheblock — APPROVED** [review]
Needs a follow up PR for edited diagram

File: docs/how-to/run-at-scale.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+At scale, deploy multiple Web3Signer instances behind a load balancer. This setup ensures balanced request distribution.
+Use an ingress load balancer to spread requests evenly across instances. This prevents overloading of single instances.
+Connect all Web3Signer instances to the same slashing database. This allows parallel signing without slashing risk.
+
+For more information, see the [Kiln article](https://www.kiln.fi/post/learnings-from-running-web3signer-at-scale-on-holesky) on running Web3Signer at scale.
+
+## Hardware recommendations
+
+Tests run nodes managing 10,000 keys on various testnets. For example, a single Azure Standard D8as v5 VM (8 vCPUs, 32 GiB memory) can host Besu, Teku, and Web3Signer simultaneously.
```
Comment (alexandratran) [inline docs/how-to/run-at-scale.md:50]:
```suggestion
The Web3Signer team runs nodes managing 10,000 keys on various testnets. For example, a single Azure Standard D8as v5 VM (8 vCPUs, 32 GiB memory) can host Besu, Teku, and Web3Signer simultaneously.
```

The first sentence was a little unclear.

File: docs/how-to/run-at-scale.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+For more information, see the [Kiln article](https://www.kiln.fi/post/learnings-from-running-web3signer-at-scale-on-holesky) on running Web3Signer at scale.
+
+## Hardware recommendations
+
+Tests run nodes managing 10,000 keys on various testnets. For example, a single Azure Standard D8as v5 VM (8 vCPUs, 32 GiB memory) can host Besu, Teku, and Web3Signer simultaneously. 
+Your specific use case may require less powerful hardware.
+
+![Dashboard for Web3Signer](../../static/img/dashboard_hw.png)
+
+Web3Signer consumes less than 2GB of JVM heap while managing 10,000 keys in this setup.
```
Comment (alexandratran) [inline docs/how-to/run-at-scale.md:55]:
```suggestion
Web3Signer consumes less than 2 GB of JVM heap while managing 10,000 keys in this setup.
```

Nitpick

File: docs/how-to/store-keys/index.md [hunk 5]
```diff
@@ -24,6 +24,8 @@ Web3Signer supports BLS12-381 or secp256k1 signing keys stored in the following
 | [YubiHSM 2]                          |     x     |  x  |
 | [USB Armory Mk II]                   |     x     |  x  |
 
+Web3Signer supports `Eth1` signing from HSMs and Vaults, but needs to load private keys into memory for `Eth2` signing.  Make sure to follow [best practices](../../get-started/key-best-practices.md) when storing private keys.
```
Comment (alexandratran) [inline docs/how-to/store-keys/index.md:27]:
```suggestion
Web3Signer supports `Eth1` signing from HSMs and vaults, but must load private keys into memory for `Eth2` signing. Follow [best practices](../../get-started/key-best-practices.md) when storing private keys.
```

File: static/img/transparent_background_diagram.png
Comment (alexandratran) [inline static/img/transparent_background_diagram.png:1]: Let's raise an issue to polish up this diagram in Figma (if it is accurate and up-to-date).

Comment (joaniefromtheblock) [reply static/img/transparent_background_diagram.png:1]: yes, noted  that in the PR and just opened an issue. Will need to get further input from eng. The source image seems to be missing info

**Pull request merged**